### ISSUE TITLE
MongoDB Database Errors

### ISSUE THREAD

**Issue by j3rkb0i** [phcollignon/helm3#151]
I am on lab6 and I cannot get the MongoDB database to use the {{ .Release.Name }}-{{ .Chart.Name }} name in the mongodb.yaml or monogdb-service.yaml.  I've been successful in converting the frontend and backend services.  I even have the database persistent volume, pvc and secret using this convention.  However, when I attempt to convert anything in mongodb.yaml or mongodb-service.yaml, the backend deployment errors out and restarts and I get the following error in the backend pod logs:

TypeError: Cannot read property 'insert' of undefined
    at /home/node/app/server.js:48:14
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/node/app/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/node/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)
    at /home/node/app/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/home/node/app/node_modules/express/lib/router/index.js:335:12)
    at next (/home/node/app/node_modules/express/lib/router/index.js:275:10)
    at /home/node/app/server.js:32:3
    at Layer.handle [as handle_request] (/home/node/app/node_modules/express/lib/router/layer.js:95:5)

I've tried Firefox and Chrome (I'm working from an Ubuntu laptop), but the error is consistent.

**Comment by j3rkb0i** [comment]
I forgot an update.  Just refreshing the page with this configuration gives the following error:

Guestbook API listening on port 3000!
/home/node/app/node_modules/mongodb/lib/topologies/server.js:240
            throw err;
            ^

Error: getaddrinfo EAI_AGAIN mongodb:27017
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:67:26)

**Comment by j3rkb0i** [comment]
One more udpate.  Lab7 final behaves the same way, but lab8 final works.

**Comment by j3rkb0i** [comment]
Last last update (head hung in shame).  In the "Customizing Frontend Chart Values" recording, I paused after you demo'd the frontend variables and said we would need to do the same for backend and database.  I was successful with backend and ran into the errors you later mention on the database deployment.  Unfortunately, I spent a great deal of time debugging and narrowing down the issue and drove myself crazy trying to figure out why I couldn't change the mongodb.yaml and mongodb-service.yaml names and metadata not related to the pv, pvc and secret.

It was only after I listened to the next 2 minutes (remember I paused ... [sigh]), that I realized the error was part of the learning curve and would be resolved in the next recording.

TLDR - I tried to work ahead and got burnt.  Please ignore this issue.
